src/component/DefaultLegendContent.tsx

```diff
@@ -42,6 +42,7 @@ export interface Payload {
   id?: string;
   type?: LegendType;
   color?: string;
+  dataKey?: string | number;
```

Comment:
If this is not correctly typed (which I can believe), how come it did not surface? Is there an any somewhere that we should also get rid off?